lib/js/src/manager/screen/menu/_MenuManagerBase.js

```diff
@@ -0,0 +1,447 @@
+    }
+
+    /**
+     * After this method finishes, the manager is ready
+     * @returns {Promise} - A promise.
+     */
+    async start () {
+        this._transitionToState(_SubManagerBase.READY);
+        await super.start();
+    }
+
+    /**
+     * Teardown method
+     */
+    dispose () {
+        this._menuCells = [];
+        this._currentMenuCells = [];
+        this._currentHmiLevel = HMILevel.HMI_NONE;
+        this._currentSystemContext = SystemContext.SYSCTXT_MAIN;
+        this._dynamicMenuUpdatesMode = DynamicMenuUpdatesMode.ON_WITH_COMPAT_MODE;
+        this._windowCapability = null;
+        this._menuConfiguration = null;
+        this._currentMenuConfiguration = null;
+        this._cancelAllTasks(); // cancel all operations
+        // remove the listeners
+        if (typeof this._hmiListener === 'function') {
+            this._lifecycleManager.removeRpcListener(FunctionID.OnHMIStatus, this._hmiListener);
+        }
+        if (typeof this._commandListener === 'function') {
+            this._lifecycleManager.removeRpcListener(FunctionID.OnCommand, this._commandListener);
+        }
+        if (typeof this._onDisplayCapabilityListener === 'function') {
+            this._lifecycleManager.removeOnSystemCapabilityListener(SystemCapabilityType.DISPLAYS, this._onDisplayCapabilityListener);
+        }
+        super.dispose();
+    }
+
+    /**
+     * Suspend the queue if the HMI level is NONE since we want to delay sending RPCs until we're in non-NONE
+     */
+    _updateTransactionQueueSuspended () {
+        if (this._currentHmiLevel === HMILevel.HMI_NONE || this._currentSystemContext === SystemContext.SYSCTXT_MENU) {
+            console.log(`MenuManagerBase - Suspending the transaction queue. Current HMI level is: ${this._currentHmiLevel}, current system context is: ${this._currentSystemContext}`);
+            this._canRunTasks = false;
+        } else {
+            console.log('MenuManagerBase - Starting the transaction queue');
+            this._canRunTasks = true;
+            this._invokeTaskQueue();
+        }
+    }
+
+    /**
+     * Sets the dynamic updates mode
+     * @param {DynamicMenuUpdatesMode} value - The dynamic updates mode
+     */
+    _setDynamicMenuUpdatesMode (value) {
+        this._dynamicMenuUpdatesMode = value;
+    }
+
+    /**
+     * Gets the dynamic updates mode
+     * @returns {DynamicMenuUpdatesMode} - The dynamic updates mode
+     */
+    _getDynamicMenuUpdatesMode () {
+        return this._dynamicMenuUpdatesMode;
+    }
+
+
+    /**
+     * Creates and sends all associated Menu RPCs
+     * @param {MenuCell[]} cells - The menu cells that are to be sent to the head unit, including their sub-cells.
+     */
+    _setMenuCells (cells = []) {
+        if (cells === null) {
+            console.error('MenuManagerBase - Cells list is null. Skipping...');
+            return;
+        }
+
+        if (!this._menuCellsAreUnique(cells.map(cell => cell.clone()), [])) {
```

Comment:
I don't think this scenario is possible because the time between the two clonings happens synchronously. Regardless, I see that cloning twice is unnecessary and so I will remove the cloning for when the cells are being checked for uniqueness.